### PR TITLE
Mod Loader 2.1.0 Release

### PR DESCRIPTION
**Pull request by MarshMello0**
The release of 2.1.0 will be ready once the website is updated. This does have some extras like the work in progress settings page, multiplayer and dev but are not shown in the release.

Features

- Launcher Auto-Updating when the user launches the game
- New Method for Update .dll using hash
- The installer now can update without affecting installed mods or skins
- URI to install mods with one click of a button on the website.

### CODE REVIEW

**Comment by MarshMello0** [comment]
As the new method of checking the mod loader dll is using hashes and the data class has been changed.
I feel like the old clients will just freeze before it says the exe needs to be updated.

Hopefully, just a post in the two discords should inform people of the release.

File: ModLoader/Dev Console/ConsoleInput.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.IO;
+
+namespace Windows
+{
+	public class ConsoleInput
+	{
+		//public delegate void InputText( string strInput );
+		public event System.Action<string> OnInputText;
```
Comment (ketkev) [inline ModLoader/Dev Console/ConsoleInput.cs]: In `System.Action<>` the `System.` isn't necessary

File: ModLoader/Dev Console/ConsoleInput.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			{
+				OnInputText( strtext );
+			}
+		}
+
+		public void Update()
+		{
+			if ( !Console.KeyAvailable ) return;
+			var key = Console.ReadKey();
+
```
Comment (ketkev) [inline ModLoader/Dev Console/ConsoleInput.cs]: These if statements could be cleaned up a bit with a switch statements but it's not necessary

File: ModLoader/ModLoaderManager.cs [hunk 3]
```diff
@@ -95,15 +119,99 @@ private void Awake()
...
+            int newVersion = 200;
+            for (int i = 0; i < args.Length; i++)
+            {
+                if (args[i].Equals("-updateLauncher"))
+                {
+                    if (int.TryParse(args[i + 1],out newVersion))
+                    {
+                        using (client = new WebClient())
+                        {
+                            Debug.Log("Downloading Launcher Update from : " + "http://localhost:81/files/updates/exe/" + newVersion.ToString() + "/VTOLVR-ModLoader.exe");
```
Comment (ketkev) [inline ModLoader/ModLoaderManager.cs]: `"http://localhost:81/files/updates/exe/"`

**Review by ketkev — CHANGES_REQUESTED** [review]
I have to type something here

**Review by ketkev — APPROVED** [review]
Looks good

**Pull request merged**